sdk/attestation/azure-security-attestation/src/test/java/com/azure/security/attestation/AttestationClientTestBase.java

```diff
@@ -0,0 +1,306 @@
+                } catch (JOSEException e) {
+                    logger.logExceptionAsError(new RuntimeException(e.toString()));
+                }
+
+
+                JWTClaimsSet claims = null;
+                try {
+                    claims = token.getJWTClaimsSet();
+                } catch (ParseException e) {
+                    logger.logExceptionAsError(new RuntimeException(e.toString()));
+                }
+                assertNotNull(claims);
+                return Mono.just(claims);
+            });
+    }
+
+    /**
+     * Create a JWS Signer from the specified PKCS8 encoded signing key.
+     * @param signingKeyBase64 Base64 encoded PKCS8 encoded RSA Private key.
+     * @return JWSSigner created over the specified signing key.
+     * @throws NoSuchAlgorithmException - should never  throws this.
+     * @throws InvalidKeySpecException - Can throw this if the key is invalid.
+     */
+    JWSSigner getJwsSigner(String signingKeyBase64) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        byte[] signingKey = Base64.getDecoder().decode(signingKeyBase64);
+        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(signingKey);
+        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+        PrivateKey privateKey = keyFactory.generatePrivate(keySpec);
+        return new RSASSASigner(privateKey);
+    }
+
+    /**
+     * Find the signing certificate associated with the specified SignedJWT token.
+     *
+     * This method depends on the token
+     * @param token - MAA generated token on which to find the certificate.
+     * @param client - Http Client used to retrieve signing certificates.
+     * @param clientUri - Base URI for the attestation client.
+     * @return X509Certificate which will have been used to sign the token.
+     */
+    Mono<X509Certificate> getSigningCertificateByKeyId(SignedJWT token, HttpClient client, String clientUri) {
+        AttestationClientBuilder builder = getBuilder(client, clientUri);
+        return builder.buildSigningCertificatesAsyncClient().get()
+            .map(keySet -> {
+                CertificateFactory cf = null;
+                try {
+                    cf = CertificateFactory.getInstance("X.509");
+                } catch (CertificateException e) {
+                    logger.logExceptionAsError(new RuntimeException(e.toString()));
+                }
+
+                String keyId = token.getHeader().getKeyID();
+
+                for (JsonWebKey key : keySet.getKeys()) {
+                    if (keyId.equals(key.getKid())) {
+                        Certificate cert = null;
+                        try {
+                            assert cf != null;
+                            cert = cf.generateCertificate(base64ToStream(key.getX5C().get(0)));
+                        } catch (CertificateException e) {
+                            logger.logExceptionAsError(new RuntimeException(e.toString()));
```

Comment:
IntelliJ suggests a lot of squiggles here. OnNext should return normally. All these should be wrapped in a `return Mono.error(<error>);`.

Reply:
Confused - first off, not seeing squiggles on my machine, not sure why. Also, not sure how I return a Mono.error() here, since Mono.error causes the lambda to return an Object, not an X509Certificate, but the whole point of this function is to either (a) return the certificate which matches the specified key ID, or (b) throw an exception.